src/Rector/ClassMethod/AddRouteAnnotationRector.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Symfony\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use Rector\BetterPhpDocParser\ValueObject\PhpDoc\DoctrineAnnotation\CurlyListNode;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Symfony\DataProvider\RouteMapProvider;
+use Rector\Symfony\PhpDocNode\SymfonyRouteTagValueNodeFactory;
+use Rector\Symfony\Tests\Rector\ClassMethod\AddRouteAnnotationRector\AddRouteAnnotationRectorTest;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see AddRouteAnnotationRectorTest
+ */
+class AddRouteAnnotationRector extends AbstractRector
+{
+    public function __construct(
+        private readonly RouteMapProvider $routeMapProvider,
+        private readonly SymfonyRouteTagValueNodeFactory $symfonyRouteTagValueNodeFactory
+    ) {
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof ClassMethod) {
+            return null;
+        }
+
+        $routeMap = $this->routeMapProvider->provide();
+        if (! $routeMap->hasRoutes()) {
+            return null;
+        }
+
+        $class = $this->betterNodeFinder->findParentType($node, Class_::class);
+        $className = $this->nodeNameResolver->getName($class);
+        $methodName = $this->nodeNameResolver->getName($node);
+
+        $fqcnAndMethodName = sprintf('%s::%s', $className, $methodName);
+        $route = $routeMap->getRouteByMethod($fqcnAndMethodName);
+
+        if ($route === null) {
+            return null;
+        }
+
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($node);
+        $routeAnnotation = $phpDocInfo->getByAnnotationClass(\Symfony\Component\Routing\Annotation\Route::class);
+
+        if ($routeAnnotation !== null) {
+            return null;
+        }
+
+        $items = [
+            'path' => $route->getPath(),
+            'name' => $route->getName(),
+        ];
+
+        $defaults = $route->getDefaults();
+        unset($defaults['_controller']);
+        if (count($defaults) > 0) {
+            $items['defaults'] = new CurlyListNode($defaults);
+        }
+
+        if ($route->getHost() !== '') {
+            $items['host'] = $route->getHost();
+        }
+
+        if (count($route->getSchemes()) > 0) {
+            $items['schemes'] = new CurlyListNode($route->getSchemes());
+        }
+
+        if (count($route->getMethods()) > 0) {
+            $items['methods'] = new CurlyListNode($route->getMethods());
+        }
+
+        if ($route->getCondition() !== '') {
+            $items['condition'] = $route->getCondition();
+        }
+
+        $phpDocInfo->addTagValueNode($this->symfonyRouteTagValueNodeFactory->createFromItems($items));
```

Comment:
So here is still my problem of the missing quotes. I've managed to add a first test to show you the problem